pysrc/Random.cpp

```diff
@@ -19,9 +19,49 @@
 
 #include "PyBind11Helper.h"
 #include "Random.h"
+#include <numpy/random/bitgen.h>  // For bitgen_t struct
+
+// For reference:
+/*
+typedef struct bitgen {
+  void *state;
+  uint64_t (*next_uint64)(void *st);
+  uint32_t (*next_uint32)(void *st);
+  double (*next_double)(void *st);
+  uint64_t (*next_raw)(void *st);
+} bitgen_t;
+*/
 
 namespace galsim {
 
+    uint32_t next_uint32(void* st)
+    {
+        BaseDeviate* rng = static_cast<BaseDeviate*>(st);
+        return (uint32_t) rng->raw();
+    }
+
+    uint64_t next_uint64(void* st)
+    {
+        uint32_t i1 = next_uint32(st);
+        uint32_t i2 = next_uint32(st);
+        return ((uint64_t) i1 << 32) + i2;
+    }
+
+    double next_double(void* st)
+    {
+        return next_uint32(st) / ((double) std::numeric_limits<uint32_t>::max()+1.);
```

Comment:
Could this use `next_uint64` instead to give a larger range of possible doubles?  Not worth the speed hit?

Reply:
I'd be surprised if any use case would benefit from using 64 bits here rather than 32.  Certainly nothing in GalSim proper needs it, as this version matches the current `ud.generate()` functionality, which is what we're (sometimes) replacing with this.  So I think it's probably not worth cutting the speed in half to get a real 64 bit range of output values.